cpp/src/copying/copy.cu

```diff
@@ -38,18 +40,40 @@ struct copy_if_else_functor_impl {
   }
 };
 
+/**
+ * @brief Functor to fetch a device-view for the specified scalar/column_view.
```

Comment:
Would `constexpr if` not work for this? I'm mostly curious if it failed more than trying to push you to it.

Reply:
I haven't been able to boil this down to something reproducible via `godbolt`, to illustrate the problem. I can try again.

Reply:
If it won't work, that's fine.

Reply:
I'm taking another crack at this.

Reply:
Nope. I can't seem to get around the warning. For the moment, I'll leave this where it is.

Reply:
The two `operator()` return different types, thus you can't use `constexpr if`.

Reply:
> The two `operator()` return different types, thus you can't use `constexpr if`.

@ttnghia, I'm not sure which `operator()` you're referring to here, but I have switched over to using `if constexpr`. 

@hyperbolic2346: The latest commit has one way to work around the warning. (Hat tip to @vuule for the advice.)